python/pyspark/sql/tests.py

```diff
@@ -3036,6 +3052,9 @@ def test_toPandas_arrow_toggle(self):
         pdf = df.toPandas()
         self.spark.conf.set("spark.sql.execution.arrow.enable", "true")
         pdf_arrow = df.toPandas()
+        # need to remove timezone for comparison
+        pdf_arrow["7_timestamp_t"] = \
+            pdf_arrow["7_timestamp_t"].apply(lambda ts: ts.tz_localize(None))
```

Comment:
Without Arrow, Spark will provide timestamps that are tz-naive.  We would need to construct the Pandas DataFrame, then call `tz_localize` with the `SESSION_LOCAL_TIMEZONE` which would then change the internal values to be normalized to UTC, as @wesm pointed out above.  I don't think this is a very good idea, because it assumes the timestamp was created with the `SESSION_LOCAL_TIMEZONE`